response_operations_ui/views/respondents.py

```diff
@@ -110,12 +154,106 @@ def respondent_details(respondent_id):
+    pending_transfer_surveys = []
+    pending_share_surveys = []
+    for pending_survey in pending_surveys:
+        if "is_transfer" in pending_survey and pending_survey["is_transfer"] is True:
+            pending_transfer_surveys.append(pending_survey)
+        else:
+            pending_share_surveys.append(pending_survey)
+    formatted_transfer_surveys = get_formatted_pending_surveys(pending_transfer_surveys)
+    formatted_share_surveys = get_formatted_pending_surveys(pending_share_surveys)
+
     return render_template(
         "respondent.html",
         respondent=respondent,
         enrolments=enrolments,
         breadcrumbs=breadcrumbs,
         mark_for_deletion=account["mark_for_deletion"],
+        pending_transfer_surveys=formatted_transfer_surveys,
+        pending_share_surveys=formatted_share_surveys,
+        respondent_id=respondent_id,
+    )
+
+
+def get_formatted_pending_surveys(pending_surveys: list):
+    """
+    Get formatted pending surveys related to the respondent
+    :param pending_surveys: pending survey list to be formatted
+    :type pending_surveys: list
+    :return: pending survey formatted list
+    :rtype: list
+    """
+    formatted_pending_surveys = []
+    if len(pending_surveys) > 0:
+        distinct_batch_number = {pending_surveys["batch_no"] for pending_surveys in pending_surveys}
+        for batch_number in distinct_batch_number:
+            business_surveys_list = []
+            distinct_businesses = set()
+            for pending_survey in pending_surveys:
+                if pending_survey["batch_no"] == batch_number:
+                    distinct_businesses.add(pending_survey["business_id"])
+                    recipient_email = pending_survey["email_address"]
+                    time_shared = pending_survey["time_shared"]
+            for business_id in distinct_businesses:
+                business_surveys = []
+                for pending_survey in pending_surveys:
+                    if pending_survey["business_id"] == business_id and pending_survey["batch_no"] == batch_number:
+                        business_surveys.append(
+                            survey_controllers.get_survey_by_id(pending_survey["survey_id"]).get("shortName")
+                        )
+                selected_business = party_controller.get_business_by_party_id(business_id)
+                business_surveys_list.append({"business_name": selected_business["name"], "surveys": business_surveys})
+            formatted_pending_surveys.append(
+                {
+                    "batch_no": batch_number,
+                    "recipient_email": recipient_email,
+                    "time_shared": convert_events_to_new_format(time_shared),
+                    "pending_survey_details": business_surveys_list,
+                }
+            )
+    return formatted_pending_surveys
+
+
+def convert_events_to_new_format(date):
```

Comment:
Can you do type hints for this function?  I assume date is an str, and the return type is also a string?  That's just a guess

Reply:
ok